### ISSUE TITLE
Import type code action leads to invalid code if used inside an inline module

### ISSUE THREAD

**Issue by jonasbb** [rust-analyzer/rust-analyzer#11585]
**rust-analyzer version**: (eg. output of "Rust Analyzer: Show RA Version" command)
rust-analyzer version: 4e72700e3 2022-02-28 stable
I tested some older versions up to 0.2.792 and could reproduce it in any of them.

**rustc version**: (eg. output of `rustc -V`)
rustc 1.61.0-nightly (4b043faba 2022-02-24)

**relevant settings**: (eg. client settings, or environment variables like `CARGO`, `RUSTUP_HOME` or `CARGO_HOME`)
```json
"rust-analyzer.assist.importGroup": false
```
This is the only setting I need to change in VS Code to trigger the bug.

Executing the `` Import `std::collections::HashMap` `` code action on `HashMap` results in invalid code. I can only reproduce this by setting `importGroup`. The default value does not trigger this bug.

Before:
```rust
mod foobar {
    fn create_map() -> HashMap<i32, i32> {
        todo!()
    }
}
```

After:
```rust
mod foobar use std::collections::HashMap;

{
    fn create_map() -> HashMap<i32, i32> {
        todo!()
    }
}
```

**Comment by Veykril** [comment]
Bug is *probably* caused in here somewhere https://github.com/rust-analyzer/rust-analyzer/blob/a55dd29e88c6f2aa37c39b790ab7036596e74686/crates/ide_db/src/helpers/insert_use.rs#L338-L348
We seem to pick the wrong insertion offset for inline modules
